### DISCUSSION TITLE
Does backup automatically delete it on S3 also?

### DISCUSSION THREAD

**Discussion by vigneshksaithal** [pocketbase/pocketbase#4577] [Q&A] [answered]
I have setup backup in Cloudflare R2 and after the limit it is not getting deleted.

**Comment by ganigeorgiev** [comment, marked as answer]
Yes, the autogenerated backups should be rotated no matter whether S3 or the local filesystem is used. 

Could you elaborate what/which backup is not getting deleted in your case?
Are the backups you expect to be deleted visible in the Admin UI settings or only in R2 dashboard?

If it is only in R2 then it is possible that R2 could have history/versioning enabled (_or Cloudflare could batch the delete operations_).

Also note that only the autogenerated backups, those starting with `@auto`, will be rotated.

**Reply by vigneshksaithal** [reply]
Earlier I had set number of backups to 50, and few days back I set it to 30(1 daily). But, there are 33 backups in Pocketbase dashboard and in R2 dashboard also. Is it due to my previous limit?

**Reply by ganigeorgiev** [reply]
I'm not sure, if you are not able to reproduce it then most likely that is the cause.

Note that the auto backup remove operation doesn't execute after the settings save but after the first new backup with the new settings.

Also make sure to use the latest PocketBase version as we had a small issue in pre v0.22 where the backup cron settings wasn't applied properly (https://github.com/pocketbase/pocketbase/discussions/4431).

**Reply by vigneshksaithal** [reply]
Oh, I got it, I have 3 @update backups, so does it mean it will delete only the @auto ones? Not the @update ones?

**Reply by ganigeorgiev** [reply]
Yes, only the backups starting with `@auto_pb_backup_...` will be auto deleted/rotated.

**Reply by vigneshksaithal** [reply]
Thank you @ganigeorgiev for answering my question.
